### ISSUE TITLE
Populate NOAA-EMC/GSI-fix with ASCII GSI fix files

### ISSUE THREAD

**Issue by RussTreadon-NOAA** [NOAA-EMC/GSI-fix#1]
GSI fix files currently reside in gerrit:GSI-fix.   This repository contains both ASCII (text) and binary GSI fix files.   The size of gerrit:GSI-fix has grown over the years.  Changes to binaries result in the generation of new copies, not incremental updates. 

The decision has been made to version control ASCII GSI fix files in NOAA-EMC/GSI-fix.   For the time being binary GSI fix files will be staged on WCOSS2 and NOAA RDPHCS machines.

This issue is opened to document the addition of ASCII (text) GSI-fix files to NOAA-EMC/GSI-fix.

**Comment by RussTreadon-NOAA** [comment]
Complete the following steps:

1. Create `feature/ascii` as a branch in gerrit:GSI-fix.   `feature/ascii` originated as a copy of `rev2` at `d118c5c`.
2. Clone rev2 and checkout a fresh copy of `feature/ascii`
3. Use `file` command to identify text and binary (data) files
4. `git rm` binary files
5. `git commit`
6. `git push origin feature/ascii`.   Done at 099bc51.

**Comment by RussTreadon-NOAA** [comment]
@aerorahul ,  gerrit:GSI-fix branch `feature/ascii` contains ASCII text GSI -fix files.   All binary files have been removed.    The head of `feature/ascii` is 099bc51.   

At your convenience would you please import gerrit:GSI-fix `feature/ascii` into NOAA-EMC/GSI-fix?

**Comment by aerorahul** [comment]
@RussTreadon-NOAA 
https://github.com/NOAA-EMC/GSI-fix has been populated with `feature/ascii`.  The branch in Github is called `develop`.

The history of all the ascii files has been retained.

In order to preserve the commit history of these ascii files, I had to remove the references to the binary files from the git history.  As a result, the hash you see in the `develop` branch [e999eb9](https://github.com/NOAA-EMC/GSI-fix/commit/e999eb95b6dcf8a94a04a1a336f5de4ee12354df), is exactly one commit before the hash of `feature/ascii` in `gerrit::GSI-fix` which physically removed all the binary files.

A `diff -q -r` between the `gerrit::GSI-fix` branch `feature/ascii` and `github::NOAA-EMC/GSI-fix` branch `develop` shows that the files are identical.
Git attributes are different and expected as I pruned all the git history of all binary files from the gerrit branch before migrating to Github.

At your convenience, please perform a comparison and let me know if you notice any irregularities.

**Comment by RussTreadon-NOAA** [comment]
Thank you, @aerorahul , for populating NOAA-EMC:GSI-fix with ASCII files from gerrit:GSI-fix.   

I cloned `NOAA-EMC/GSI-fix` on Hera.  A comparison of files in `NOAA-EMC/GSI-fix/develop` and `NOAA-EMC/GSI/develop/fix` did not find any irregularities.   `NOAA-EMC/GSI-fix/develop` contains ASCII fix files consistent with what we find in `NOAA-EMC/GSI/develop/fix` (which, in turn, is a snapshot of `gerrit:GSI-fix/rev2`).  `NOAA-EMC/GSI-fix/develop` does not contain binary files.   This is correct.

The goal of this issue has been achieved.   I think we can close this issue.


The next step (next issue) is to explore how best to modify the `fix` submodule in `NOAA-EMC/GSI/develop`.   Ideally, a future `git clone --recursive` of `NOAA-EMC/GSI` would 

1. populate `fix` with ASCII GSI fix files from a `NOAA-EMC/GSI-fix` hash 
2. populate `fix` with binary GSI fix files by either
   - creating links to binary GSI fix files in a staged location on the given machine
   - downloading binary GSI fix files from a central server (e.g., emcrzdm)

If the above is possible, the user will effectively see no change in local working copies of `fix` when cloning `NOAA-EMC/GSI`.  This allows existing scripts to continue working _as is_.

**Comment by aerorahul** [comment]
@RussTreadon-NOAA 
Thank you for confirming the files are identical.
You may close this issue.

As for the next step (next issue), if possible, could you provide a list of the binary files that need to be hosted elsewhere and copied/linked/checked out into the `NOAA-EMC::GSI/fix` space? 
For 2., I have a third suggestion where the binary files are checked out/linked during the build process.
I can illustrate this better when I have the list of binary files.

Before we do the next steps, I would suggest we:
1. lock `develop` in GSI-fix such that new commits will require PR's.
2. add a `.gitignore` (to prevent users from accidentally adding unnecessary artifacts), `README.md` (to describe this repository and its contents so users understand), and a `LICENSE` file (LGPL license just like other NOAA-EMC and ufs-community repos).

**Comment by RussTreadon-NOAA** [comment]
@aerorahul ,

For suggestion 1, @KateFriedman-NOAA maintains copies of GSI fix files in `$FIX_DIR/gsi/$gsi_ver` with `$FIX_DIR` defined on various machines in g-w `sorc/link_workflow.sh`
```
#------------------------------
#--model fix fields
#------------------------------
case "${machine}" in
  "wcoss2")   FIX_DIR="/lfs/h2/emc/global/noscrub/emc.global/FIX/fix" ;;
  "hera")     FIX_DIR="/scratch1/NCEPDEV/global/glopara/fix" ;;
  "orion")    FIX_DIR="/work/noaa/global/glopara/fix" ;;
  "jet")      FIX_DIR="/lfs4/HFIP/hfv3gfs/glopara/git/fv3gfs/fix" ;;
  "s4")       FIX_DIR="/data/prod/glopara/fix" ;;
```
and `$gsi_ver=20221128` in g-w `versions/fix.ver`.

`file * | grep -i -v text` in `$FIX_DIR/gsi/$gsi_ver` returns a list of binary GSI fix files.   This command also returns `Big_Endian` and `gfsv16_historical` as directories.   `Big_Endian` only contains binary fix files.   `gfsv16_historical` only contains ASCII fix files.

For suggestion 2,
- I am willing to lock `develop` but do not know how to do this.   
- Is it sufficient to add the NOAA-EMC/GSI `.gitignore` to `GSI-fix`?
- I will add a simple `README.md` similar to what we have for [GSI-utils](https://github.com/NOAA-EMC/GSI-utils/blob/develop/README.md)
- I see various LICENSE files in NOAA-EMC repos.   GDASApp has [LICENSE](https://github.com/NOAA-EMC/GDASApp/blob/develop/LICENSE).   UPP has [LICENSE.md](https://github.com/NOAA-EMC/UPP/blob/develop/LICENSE.md), NCEPLIBS-bufr has a different [LICENSE.md](https://github.com/NOAA-EMC/NCEPLIBS-bufr/blob/develop/LICENSE.md).   Which example should I follow?

I see that some of the above is obsolete given GSI-fix issue #2 and PR [#3](https://github.com/NOAA-EMC/GSI-fix/pull/3).

I'll close this issue with this comment and move onto PR [#3](https://github.com/NOAA-EMC/GSI-fix/pull/3).
